src/wp-includes/block-supports/elements.php

```diff
@@ -97,40 +98,27 @@ function wp_render_elements_support( $block_content, $block ) {
  * @return null
  */
 function wp_render_elements_support_styles( $pre_render, $block ) {
-	$block_type                    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
-	$skip_link_color_serialization = wp_should_skip_block_supports_serialization( $block_type, 'color', 'link' );
-	if ( $skip_link_color_serialization ) {
-		return null;
-	}
-
-	$link_color = null;
-	if ( ! empty( $block['attrs'] ) ) {
-		$link_color = _wp_array_get( $block['attrs'], array( 'style', 'elements', 'link', 'color', 'text' ), null );
-	}
+	$block_type           = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$element_block_styles = isset( $block['attrs']['style']['elements'] ) ? $block['attrs']['style']['elements'] : null;
 
 	/*
 	* For now we only care about link color.
-	* This code in the future when we have a public API
-	* should take advantage of WP_Theme_JSON::compute_style_properties
-	* and work for any element and style.
 	*/
-	if ( null === $link_color ) {
-		return null;
-	}
-
-	$class_name = wp_get_elements_class_name( $block );
+	$skip_link_color_serialization = wp_should_skip_block_supports_serialization( $block_type, 'color', 'link' );
 
-	if ( strpos( $link_color, 'var:preset|color|' ) !== false ) {
-		// Get the name from the string and add proper styles.
-		$index_to_splice = strrpos( $link_color, '|' ) + 1;
-		$link_color_name = substr( $link_color, $index_to_splice );
-		$link_color      = "var(--wp--preset--color--$link_color_name)";
+	if ( $skip_link_color_serialization ) {
+		return null;
 	}
-	$link_color_declaration = esc_html( safecss_filter_attr( "color: $link_color" ) );
-
-	$style = ".$class_name a{" . $link_color_declaration . ';}';
-
-	wp_enqueue_block_support_styles( $style );
+	$class_name        = wp_get_elements_class_name( $block );
+	$link_block_styles = isset( $element_block_styles['link'] ) ? $element_block_styles['link'] : null;
+
+	wp_style_engine_get_styles(
+		$link_block_styles,
+		array(
+			'selector' => ".$class_name a",
+			'context'  => 'block-supports',
+		)
+	);
```

Comment:
Don't we have to do something with the return value? 🤔

Reply:
I tried a paragraph block with a link and the link color set to orange, and it really isn't reflected on the frontend:

![image](https://user-images.githubusercontent.com/96308/191018739-56690472-2c0e-44f6-b772-a1b4c5ac3d76.png)

![image](https://user-images.githubusercontent.com/96308/191018884-ec89857b-5324-4b47-87fb-9158ab0053fe.png)

Reply:
Ah, looks like this needs #3273 in order for the relevant styling to be added!

Reply:
> Don't we have to do something with the return value? 🤔

Not here since the style engine will register styles in a store using the context as key. Gutenberg fetches stored styles at render time and prints them out.